pkg/destroy/gcp/cloudcontroller.go

```diff
@@ -121,7 +118,10 @@ func (o *ClusterUninstaller) discoverCloudControllerLoadBalancerResources(ctx co
 	loadBalancerFilterFunc := o.createLoadBalancerFilterFunc(loadBalancerName)
 
 	// Discover associated addresses: loadBalancerName
-	found, err := o.listAddressesWithFilter(ctx, regionalAddressResource, "items(name),nextPageToken", loadBalancerFilterFunc)
+	found, err := o.listAddressesWithFilter(ctx, regionalAddressResource, "items(name),nextPageToken", func(item *compute.Address) bool {
+		// address should have labels but none observed
+		return strings.HasPrefix(item.Name, loadBalancerName)
+	})
```

Comment:
This has the same check as `loadBalancerFilterFunc` right? Can we keep the original implementation? We can still keep the comment though.